promise_test.go

```diff
@@ -249,3 +249,147 @@ func TestPromise_All3(t *testing.T) {
 		return
 	}
 }
+
```

Comment:
Also for my tests and their format - The format is really clear (test-case table format) and is a better approach to testing since you can arbitrarily update the tests cases list easily, also my tests take care of a couple of cases current test cases do not :).

Reply:
Yep, looks good to be honest.
Maybe after you're done with the feature PRs we can work on refactoring older tests as well

Reply:
Jehjeh *nods in confirmation*, agreed :) :+1: